# Counts modeling {#model}

In this section, I will discuss the statistical models that are
often used to analyze RNA-seq data, in particular gene-level count
matrices. I will then use the *DESeq2* package to calculate
scaling factors, estimate biological dispersion within groups of
samples, and perform differential testing per gene [@Love2014].

Some other popular Bioconductor packages for RNA-seq analysis include the 
*edgeR* package [@Robinson2010,@McCarthy2012] and the *limma-voom*
method in the *limma* package [@Law2014]. The approach taken by
*DESeq2* for estimation of dispersion is similar to the method
proposed by @Wu2012 in the *DSS* Bioconductor package.

* Counts, sequencing depth, precision
* Marginal testing, multiple correction, IHW

```{r colsums}
assayNames(gse)
cs <- colSums(assay(gse, "counts"))
hist(cs/1e6, col="grey", border="white")
```

```{r make-prob}
colData(gse)
cts <- assay(gse, "counts")[,c(1,4)]
idx <- rowSums(cts >= 5) == 2
cts <- cts[idx,]
p <- sweep(cts, 2, colSums(cts), "/")
```

```{r maplot}
library(rafalib)
maplot(log10(p[,1]), log10(p[,2]), n=nrow(p), 
       cex=.3, col=rgb(0,0,0,.2))
abline(h=0, col=rgb(0,0,1,.5))
```

```{r mean-prob}
mean.log.p <- rowMeans(log10(p))
hist(mean.log.p,  col="grey", border="white")
```

```{r maplot2}
library(rafalib)
maplot(log10(p[,1]), log10(p[,2]), n=nrow(p),
       xlim=c(-6.5, -3.5), ylim=c(-2,2),
       cex=.3, col=rgb(0,0,0,.2))
abline(h=0, col=rgb(0,0,1,.5))
```

```{r lfc-hist}
lfc <- log2(p[,2]) - log2(p[,1])
hist(lfc[between(mean.log.p, -6, -4)], 
     breaks=seq(-10,10,by=.1),
     xlim=c(-5,5),
     col="grey50", border="white")
```

```{r sim-maplot}
sim.cts <- matrix(rpois(nrow(p) * 2, 30e6 * p[,1]), ncol=2)
colSums(sim.cts)
idx <- rowSums(sim.cts >= 5) == 2
sim.cts <- sim.cts[idx,]
sim.p <- sweep(sim.cts, 2, colSums(sim.cts), "/")
maplot(log10(sim.p[,1]), log10(sim.p[,2]), n=nrow(p), 
       cex=.3, col=rgb(0,0,0,.2))
abline(h=0, col=rgb(0,0,1,.5))
```

```{r make-dds}
library(DESeq2)
dds <- DESeqDataSet(gse, ~line + line:condition)
keep <- rowSums(counts(dds) >= 10) >= 3
table(keep)
dds <- dds[keep,]
```

```{r deseq, cache=TRUE}
dds <- DESeq(dds)
```

```{r disp-ests}
plotDispEsts(dds, ylim=c(1e-3, .5), xlim=c(5,1e5))
```

```{r pca}
vsd <- vst(dds, blind=FALSE)
plotPCA(vsd, intgroup=c("line","condition"))
```

```{r results}
resultsNames(dds)
res <- results(dds, name="lineOCT4.conditiontrt")
summary(res)
```

```{r apeglm, cache=TRUE}
library(apeglm)
lfc <- lfcShrink(dds, coef="lineOCT4.conditiontrt", type="apeglm")
```


```{r ma-shrink}
plotMA(res, ylim=c(-4,4), 
       colNonSig="grey60", colSig="blue", colLine="grey40")
```

```{r ma-shrinkn2}
plotMA(lfc, ylim=c(-4,4), 
       colNonSig="grey60", colSig="blue", colLine="grey40")
```


```{r symbols}
lfc$SYMBOL <- mcols(dds)$SYMBOL
tab <- lfc %>% as.data.frame %>%
        filter(between(baseMean, 1e4, 1e5), 
               between(abs(log2FoldChange), 1, 4))
tab <- tab[complete.cases(tab),]
```

```{r ma-symbols}
plotMA(lfc, ylim=c(-4,4),
       colNonSig="grey60", colSig="blue", colLine="grey40")
with(tab, {
        points(baseMean, log2FoldChange, cex=2, col="blue")
        text(baseMean, log2FoldChange, SYMBOL, pos=4, col="blue")
})
```

```{r signif}
dat <- as.data.frame(lfc)
dat <- dat[complete.cases(dat),]
dat <- dat %>% mutate(sig = ifelse(padj < .1, "Y", "N"))
tab$sig <- "Y"
```

```{r ma-repel}
library(ggplot2)
library(ggrepel)
ggplot(dat, aes(baseMean, log2FoldChange, col=sig, label=SYMBOL)) + 
        geom_point() + scale_x_log10() + 
        geom_hline(yintercept=0, col="grey40") + 
        scale_color_manual(values=c("grey60", "blue")) +
        geom_point(data=tab, shape=1, size=5, show.legend=FALSE) + 
        geom_label_repel(data=tab,
                         nudge_x = 1,
                         nudge_y = 2* sign(tab$log2FoldChange),
                         show.legend=FALSE)
```

```{r ihw, cache=TRUE}
library(IHW)
res.ihw <- results(dds, name="lineOCT4.conditiontrt", filterFun=ihw)
summary(res.ihw)
```

```{r ihw-obj}
ihw.obj <- metadata(res.ihw)$ihwResult
plot(ihw.obj)
```

```{r compare-fdr}
table(res$padj < .05, IHW=res.ihw$padj < .05)
```
