<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Quantification | Statistical Modeling of High Dimensional Counts</title>
  <meta name="description" content="This book covers some basic routines for statistical modeling of high dimensional count data in using R/Bioconductor." />
  <meta name="generator" content="bookdown 0.16 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Quantification | Statistical Modeling of High Dimensional Counts" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book covers some basic routines for statistical modeling of high dimensional count data in using R/Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Quantification | Statistical Modeling of High Dimensional Counts" />
  
  <meta name="twitter:description" content="This book covers some basic routines for statistical modeling of high dimensional count data in using R/Bioconductor." />
  

<meta name="author" content="Michael I. Love" />


<meta name="date" content="2020-01-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="model.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Stat. Model. of High Dim. Counts</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this online book</a></li>
<li class="chapter" data-level="1" data-path="quantification.html"><a href="quantification.html"><i class="fa fa-check"></i><b>1</b> Quantification</a></li>
<li class="chapter" data-level="2" data-path="model.html"><a href="model.html"><i class="fa fa-check"></i><b>2</b> Counts modeling</a><ul>
<li class="chapter" data-level="2.1" data-path="model.html"><a href="model.html#a-first-exploration-of-counts"><i class="fa fa-check"></i><b>2.1</b> A first exploration of counts</a></li>
<li class="chapter" data-level="2.2" data-path="model.html"><a href="model.html#modeling-counts-with-deseq2"><i class="fa fa-check"></i><b>2.2</b> Modeling counts with DESeq2</a></li>
<li class="chapter" data-level="2.3" data-path="model.html"><a href="model.html#hypothesis-weighting"><i class="fa fa-check"></i><b>2.3</b> Hypothesis weighting</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="transcripts.html"><a href="transcripts.html"><i class="fa fa-check"></i><b>3</b> Transcript expression</a></li>
<li class="chapter" data-level="4" data-path="limits.html"><a href="limits.html"><i class="fa fa-check"></i><b>4</b> Limitations and extensions</a></li>
<li class="chapter" data-level="5" data-path="session.html"><a href="session.html"><i class="fa fa-check"></i><b>5</b> Session information</a></li>
<li class="chapter" data-level="6" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>6</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Modeling of High Dimensional Counts</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quantification" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Quantification</h1>
<p>One key initial step in analyzing RNA-seq data is to quantify,
or estimate, the number of fragments in the experiment that
can be assigned to each feature, whether a gene or a transcript
(an isoform of a gene). Alongside quantification, it is strongly
recommended to perform quality control (QC) checks on the sequence
files. Reports spanning multiple samples can be generated with
the <a href="https://multiqc.info/">MultiQC</a> software <span class="citation">(Ewels et al. <a href="#ref-Ewels2016">2016</a>)</span>. Here
I will not start with quality control checks, but instead move
straight to quantification and import of estimated counts into R,
although an example quality control report can be found in the
<em>airway2</em> package on GitHub:
<a href="https://github.com/mikelove/airway2/blob/master/inst/extdata/multiqc_report.html">MultiQC report</a>.</p>
<p>The data I will examine first in this section is from an experiment
of the effect of knocking down OCT4 and BRG1 in mouse embryos <span class="citation">(King and Klose <a href="#ref-King2017">2017</a>)</span>.
In particular, I will examine the treatment effect of knocking down OCT4.
The experiment had four groups of samples, each replicated in
triplicate. The RNA-seq data from the experiment is available in the
<em>oct4</em> Bioconductor package.</p>
<p>I will use the <em>Salmon</em> software for estimating transcript abundance <span class="citation">(Patro et al. <a href="#ref-Patro2017">2017</a>)</span>.
Briefly, <em>Salmon</em> uses the sequenced reads and the reference transcripts,
and constructs a generative model for the observed data, which includes
modeling of various technical biases that are commonly observed in RNA-seq.
<em>Salmon</em> then outputs the estimated counts for each transcript, and an “effective length”
of the transcript, which is shorter than the full length of the transcript
if it was biased to having fewer reads due to technical artifact, and longer
than the full length of the transcript if it was biased to having more reads.
For details about the Salmon method and software, refer to the
<a href="https://combine-lab.github.io/salmon/">Salmon website</a>.</p>
<p>All 12 samples from the experiment were quantified using <em>Salmon</em>.
The jobs for processing the reads with <em>Salmon</em> were executed via
Snakemake <span class="citation">(Köster and Rahmann <a href="#ref-Koster2012">2012</a>)</span>, a convenient tool for scheduling and
executing repetitive rule-based operations on input data.
The exact lines of code can be seen in the
<a href="https://github.com/mikelove/oct4/blob/master/inst/scripts/Snakefile">Snakefile</a>.</p>
<p>Most of the packages shown in this online book live within the Bioconductor
project <span class="citation">(Huber et al. <a href="#ref-Huber2015">2015</a>)</span>. Bioconductor objects are more complex than
basic objects in R, for example <em>numeric</em>, <em>character</em>, or other
simple objects, in that they often have attached metadata, such as
additional information about rows and columns, or other metadata.
We will see how to make use of the metadata throughout the various
sections by examining, e.g. <code>colData</code> for information about the columns
of a matrix, or <code>mcols</code> for metadata columns.</p>
<p>I begin by loading some data in the Bioconductor package <em>oct4</em>.
Note that this step is not useful for a typical RNA-seq workflow,
as the data will not be contained in an R package, but contained
in some directory on a server or compute cluster. So in lieu
of the <code>system.file</code> command below, which is used here to locate
a file within an R package, you could just specify the <code>dir</code>
variable to be a path to the files, e.g. <code>/path/to/data/dir</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">dir &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="dt">package=</span><span class="st">&quot;oct4&quot;</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">list.files</span>(dir)</a></code></pre></div>
<pre><code>## [1] &quot;coldata.csv&quot;     &quot;list&quot;            &quot;quants&quot;         
## [4] &quot;SraRunTable.txt&quot;</code></pre>
<p>I will use the <em>readr</em> and <em>dplyr</em> packages to read in a CSV
file with information about the samples. Because we are typically
working with “tall” count matrices, with the rows representing
genomic features such as genes or transcripts, and columns
representing samples, the sample information is referred to
as the column data or <code>coldata</code>. For more information on
dplyr, see the excellent
<a href="https://dplyr.tidyverse.org/">dplyr online documentation</a>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(readr)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">coldata &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="kw">file.path</span>(dir,<span class="st">&quot;coldata.csv&quot;</span>))</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   names = col_character(),
##   line = col_character(),
##   condition = col_character()
## )</code></pre>
<p>I next set the <code>levels</code> of the <code>line</code> and <code>condition</code> factor variables
so that <code>OCT4</code> and <code>untrt</code> are the reference levels, and I specify
the path to the quantification files I want to read in.</p>
<p>In the last step of the <code>mutate</code> call, note I specify paths to the
quantification files <code>quant.sf.gz</code>. Typically, these files are not
gzipped, but I have compressed the <code>quant.sf</code> files to reduce the
size of the data package. Note also that, although I point only to the
<code>quant.sf</code> files, the entire directory containing that file
is required for proper import of the quantification data.
There are other metadata files that provide important information
about the experiment, including uncertainty information. The
metadata files enable automatic identification of the feature set,
a computational reproducibility feature that will be described later
in this section.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(dplyr))</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">coldata &lt;-<span class="st"> </span>coldata <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">line=</span><span class="kw">factor</span>(line, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;OCT4&quot;</span>,<span class="st">&quot;BRG1&quot;</span>)),</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">         <span class="dt">condition=</span><span class="kw">factor</span>(condition, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;untrt&quot;</span>,<span class="st">&quot;trt&quot;</span>)),</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">         <span class="dt">files=</span><span class="kw">file.path</span>(dir, <span class="st">&quot;quants&quot;</span>, names, <span class="st">&quot;quant.sf.gz&quot;</span>))</a></code></pre></div>
<p>All the files exist at the locations I specified, which in this case
have the pattern <code>/&lt;DIR&gt;/quants/&lt;NAMES&gt;/quant.sf.gz</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">all</span>(<span class="kw">file.exists</span>(coldata<span class="op">$</span>files))</a></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>I will now use the Bioconductor package <em>tximeta</em> <span class="citation">(Love et al. <a href="#ref-Love2019">2019</a>)</span> to
read in the quantification data, and create a <em>SummarizedExperiment</em>
object. The <em>SummarizedExperiment</em> class is described in <span class="citation">Huber et al. (<a href="#ref-Huber2015">2015</a>)</span>,
in particular diagrammed in
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4509590/figure/F2/">this figure</a>.
Here, the <code>tximeta</code> function perform a number of operations on
behalf of the user, identifying the reference transcripts that were used
to quantify the RNA-seq reads, automatically downloading (or loading)
the relevant genomic locations of the transcripts, and attaching the
relevant genomic context (the genome version and chromosome names and
lengths). <em>tximeta</em> can perform these operations by leveraging the
hash signature of the sequence of the transcripts, which is stored
when processing bulk RNA-seq with <em>Salmon</em>, or when processing single
cell RNA-seq with the <em>alevin</em> software <span class="citation">(A. Srivastava, Malik, Smith, et al. <a href="#ref-Srivastava2019">2019</a>)</span>
(distributed with <em>Salmon</em>).</p>
<p>A small note: I specify <code>dropInfReps=TRUE</code>, because I will be
performing gene-level analysis, and I won’t make use of the
uncertainty information in this first analysis. I will show
in a later section how to make use of the inferential replicates
computed by <em>Salmon</em> during quantification.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">library</span>(tximeta)</a></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(SummarizedExperiment))</a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">se &lt;-<span class="st"> </span><span class="kw">tximeta</span>(coldata, <span class="dt">dropInfReps=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## importing quantifications</code></pre>
<pre><code>## reading in files with read_tsv</code></pre>
<pre><code>## 1 2 3 4 5 6 7 8 9 10 11 12 
## found matching transcriptome:
## [ GENCODE - Mus musculus - release M20 ]
## loading existing TxDb created: 2019-07-01 19:31:02
## loading existing transcript ranges created: 2019-09-12 12:45:55
## fetching genome info for GENCODE</code></pre>
<p>In the above output, we can see that <code>tximeta</code> identified that
the mouse reference transcripts, release M20, were used from
GENCODE <span class="citation">(Frankish, GENCODE-consoritum, and Flicek <a href="#ref-GENCODE">2018</a>)</span>. This information would be automatically
identified, even if the group that performed the quantification
did not document this when uploading processed expression data to a
public repository.</p>
<p>We can see the matrices of data that have been compiled, and
examine the genomic locations of the features (rows):</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">assayNames</span>(se)</a></code></pre></div>
<pre><code>## [1] &quot;counts&quot;    &quot;abundance&quot; &quot;length&quot;</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">rowRanges</span>(se)</a></code></pre></div>
<pre><code>## GRanges object with 137271 ranges and 3 metadata columns:
##                        seqnames          ranges strand |
##                           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |
##   ENSMUST00000193812.1     chr1 3073253-3074322      + |
##   ENSMUST00000082908.1     chr1 3102016-3102125      + |
##   ENSMUST00000162897.1     chr1 3205901-3216344      - |
##   ENSMUST00000159265.1     chr1 3206523-3215632      - |
##   ENSMUST00000070533.4     chr1 3214482-3671498      - |
##                    ...      ...             ...    ... .
##   ENSMUST00000082419.1     chrM     13552-14070      - |
##   ENSMUST00000082420.1     chrM     14071-14139      - |
##   ENSMUST00000082421.1     chrM     14145-15288      + |
##   ENSMUST00000082422.1     chrM     15289-15355      + |
##   ENSMUST00000082423.1     chrM     15356-15422      - |
##                            tx_id              gene_id
##                        &lt;integer&gt;      &lt;CharacterList&gt;
##   ENSMUST00000193812.1         1 ENSMUSG00000102693.1
##   ENSMUST00000082908.1         2 ENSMUSG00000064842.1
##   ENSMUST00000162897.1      4203 ENSMUSG00000051951.5
##   ENSMUST00000159265.1      4204 ENSMUSG00000051951.5
##   ENSMUST00000070533.4      4205 ENSMUSG00000051951.5
##                    ...       ...                  ...
##   ENSMUST00000082419.1    138833 ENSMUSG00000064368.1
##   ENSMUST00000082420.1    138834 ENSMUSG00000064369.1
##   ENSMUST00000082421.1    138825 ENSMUSG00000064370.1
##   ENSMUST00000082422.1    138826 ENSMUSG00000064371.1
##   ENSMUST00000082423.1    138835 ENSMUSG00000064372.1
##                                     tx_name
##                                 &lt;character&gt;
##   ENSMUST00000193812.1 ENSMUST00000193812.1
##   ENSMUST00000082908.1 ENSMUST00000082908.1
##   ENSMUST00000162897.1 ENSMUST00000162897.1
##   ENSMUST00000159265.1 ENSMUST00000159265.1
##   ENSMUST00000070533.4 ENSMUST00000070533.4
##                    ...                  ...
##   ENSMUST00000082419.1 ENSMUST00000082419.1
##   ENSMUST00000082420.1 ENSMUST00000082420.1
##   ENSMUST00000082421.1 ENSMUST00000082421.1
##   ENSMUST00000082422.1 ENSMUST00000082422.1
##   ENSMUST00000082423.1 ENSMUST00000082423.1
##   -------
##   seqinfo: 22 sequences (1 circular) from mm10 genome</code></pre>
<p>Because the <code>tximeta</code> function has identified the correct
reference transcripts that were used for quantification,
and stored this as metadata in the <em>SummarizedExperiment</em>
object, I can obtain additional information, such as the
correspondence of transcripts to genes. I can then perform a
summarization task, without having to look up this information
manually.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">gse &lt;-<span class="st"> </span><span class="kw">summarizeToGene</span>(se)</a></code></pre></div>
<pre><code>## loading existing TxDb created: 2019-07-01 19:31:02</code></pre>
<pre><code>## obtaining transcript-to-gene mapping from TxDb</code></pre>
<pre><code>## loading existing gene ranges created: 2019-07-01 19:32:20</code></pre>
<pre><code>## summarizing abundance</code></pre>
<pre><code>## summarizing counts</code></pre>
<pre><code>## summarizing length</code></pre>
<p>Gene-level summarization of transcript-level quantification data
is described in the paper for <em>tximport</em> <span class="citation">(Soneson, Love, and Robinson <a href="#ref-Soneson2015">2015</a>)</span>.
The <code>tximeta</code> function internally calls methods from the
<em>tximport</em> package during its operation. Note that <em>tximport</em>
provides similar functionality to <em>tximeta</em>, but instead of
returning a rich Bioconductor object, <em>tximport</em> returns
a simple list of matrices.</p>
<p>Below I show some examples of operations that can be easily
performed because I have a <em>SummarizedExperiment</em> object
with the appropriate gene ranges attached. I can easily
subset the genes (rows) based on overlaps with a given genomic range
using standard square bracket indexing. The range-based
class system and set of methods for operating on genomic ranges
is provided by the <em>GenomicRanges</em> package, which has useful
help pages and vignettes <span class="citation">(Lawrence et al. <a href="#ref-Lawrence2013">2013</a>)</span>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">rowRanges</span>(gse)</a></code></pre></div>
<pre><code>## GRanges object with 53697 ranges and 1 metadata column:
##                         seqnames              ranges strand |
##                            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |
##    ENSMUSG00000000001.4     chr3 108107280-108146146      - |
##   ENSMUSG00000000003.15     chrX   77837901-77853623      - |
##   ENSMUSG00000000028.15    chr16   18780447-18811987      - |
##   ENSMUSG00000000031.16     chr7 142575529-142578143      - |
##   ENSMUSG00000000037.16     chrX 161117193-161258213      + |
##                     ...      ...                 ...    ... .
##    ENSMUSG00000117651.1    chr17   32731010-32731806      + |
##    ENSMUSG00000117652.1    chr18     6910459-6936621      - |
##    ENSMUSG00000117653.1    chr17   94811611-94812922      - |
##    ENSMUSG00000117654.1    chr17   32863662-32877942      - |
##    ENSMUSG00000117655.1    chr19   57497390-57512784      + |
##                                       gene_id
##                                   &lt;character&gt;
##    ENSMUSG00000000001.4  ENSMUSG00000000001.4
##   ENSMUSG00000000003.15 ENSMUSG00000000003.15
##   ENSMUSG00000000028.15 ENSMUSG00000000028.15
##   ENSMUSG00000000031.16 ENSMUSG00000000031.16
##   ENSMUSG00000000037.16 ENSMUSG00000000037.16
##                     ...                   ...
##    ENSMUSG00000117651.1  ENSMUSG00000117651.1
##    ENSMUSG00000117652.1  ENSMUSG00000117652.1
##    ENSMUSG00000117653.1  ENSMUSG00000117653.1
##    ENSMUSG00000117654.1  ENSMUSG00000117654.1
##    ENSMUSG00000117655.1  ENSMUSG00000117655.1
##   -------
##   seqinfo: 22 sequences (1 circular) from mm10 genome</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">x &lt;-<span class="st"> </span><span class="kw">GRanges</span>(<span class="st">&quot;chr1&quot;</span>, <span class="kw">IRanges</span>(<span class="fl">10e6</span>, <span class="fl">11e6</span>))</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">gse[gse <span class="op">%over%</span><span class="st"> </span>x, ]</a></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 18 12 
## metadata(6): tximetaInfo quantInfo ... txomeInfo txdbInfo
## assays(3): counts abundance length
## rownames(18): ENSMUSG00000025916.10 ENSMUSG00000025917.9
##   ... ENSMUSG00000103448.1 ENSMUSG00000103810.1
## rowData names(1): gene_id
## colnames(12): SRX2236945 SRX2236946 ... SRX2236955
##   SRX2236956
## colData names(3): names line condition</code></pre>
<p>I can likewise easily subset the samples by referring to
the <code>colData</code> columns, e.g. <code>colData(gse)$line</code>.
A shortcut for referring to <code>colData</code> columns is to just use
the dollar sign directly on the object, <code>gse$line</code>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">gse[, gse<span class="op">$</span>line <span class="op">==</span><span class="st"> &quot;OCT4&quot;</span>]</a></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 53697 6 
## metadata(6): tximetaInfo quantInfo ... txomeInfo txdbInfo
## assays(3): counts abundance length
## rownames(53697): ENSMUSG00000000001.4
##   ENSMUSG00000000003.15 ... ENSMUSG00000117654.1
##   ENSMUSG00000117655.1
## rowData names(1): gene_id
## colnames(6): SRX2236945 SRX2236946 ... SRX2236949
##   SRX2236950
## colData names(3): names line condition</code></pre>
<p>Finally, I demonstrate that it is easy to add alternative
identifiers, making use of the organism annotation packages
in Bioconductor. For example, to add gene symbols, I can
use <em>tximeta</em>’s <code>addIds</code> function, and specify the <code>SYMBOL</code>
column.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="kw">library</span>(org.Mm.eg.db)</a>
<a class="sourceLine" id="cb31-2" data-line-number="2">gse &lt;-<span class="st"> </span><span class="kw">addIds</span>(gse, <span class="dt">column=</span><span class="st">&quot;SYMBOL&quot;</span>)</a></code></pre></div>
<pre><code>## mapping to new IDs using &#39;org.Mm.eg.db&#39; data package
## if all matching IDs are desired, and &#39;1:many mappings&#39; are reported,
## set multiVals=&#39;list&#39; to obtain all the matching IDs</code></pre>
<pre><code>## it appears the rows are gene IDs, setting &#39;gene&#39; to TRUE</code></pre>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and
## columns</code></pre>
<p>To check all the available columns for an organism package,
use the <code>columns</code> function:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="kw">columns</span>(org.Mm.eg.db)</a></code></pre></div>
<pre><code>##  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot; 
##  [5] &quot;ENSEMBLTRANS&quot; &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;    
##  [9] &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;     &quot;GO&quot;           &quot;GOALL&quot;       
## [13] &quot;IPI&quot;          &quot;MGI&quot;          &quot;ONTOLOGY&quot;     &quot;ONTOLOGYALL&quot; 
## [17] &quot;PATH&quot;         &quot;PFAM&quot;         &quot;PMID&quot;         &quot;PROSITE&quot;     
## [21] &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UNIGENE&quot;      &quot;UNIPROT&quot;</code></pre>

</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Ewels2016">
<p>Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016. “MultiQC: summarize analysis results for multiple tools and samples in a single report.” <em>Bioinformatics</em> 32 (19): 3047–8. <a href="https://doi.org/10.1093/bioinformatics/btw354">https://doi.org/10.1093/bioinformatics/btw354</a>.</p>
</div>
<div id="ref-GENCODE">
<p>Frankish, A, GENCODE-consoritum, and P Flicek. 2018. “GENCODE reference annotation for the human and mouse genomes.” <em>Nucleic Acids Research</em>. <a href="https://doi.org/10.1093/nar/gky955">https://doi.org/10.1093/nar/gky955</a>.</p>
</div>
<div id="ref-Huber2015">
<p>Huber, Wolfgang, Vincent J. Carey, Robert Gentleman, Simon Anders, Marc Carlson, Benilton S. Carvalho, Hector C. Bravo, et al. 2015. “Orchestrating high-throughput genomic analysis with Bioconductor.” <em>Nature Methods</em> 12 (2): 115–21. <a href="https://doi.org/10.1038/nmeth.3252">https://doi.org/10.1038/nmeth.3252</a>.</p>
</div>
<div id="ref-King2017">
<p>King, Hamish W, and Robert J Klose. 2017. “The Pioneer Factor Oct4 Requires the Chromatin Remodeller Brg1 to Support Gene Regulatory Element Function in Mouse Embryonic Stem Cells.” Edited by Irwin Davidson. <em>eLife</em> 6 (March): e22631. <a href="https://doi.org/10.7554/eLife.22631">https://doi.org/10.7554/eLife.22631</a>.</p>
</div>
<div id="ref-Koster2012">
<p>Köster, Johannes, and Sven Rahmann. 2012. “Snakemake—a Scalable Bioinformatics Workflow Engine.” <em>Bioinformatics</em> 28 (19). Oxford University Press: 2520–2. <a href="https://doi.org/10.1093/bioinformatics/bts480">https://doi.org/10.1093/bioinformatics/bts480</a>.</p>
</div>
<div id="ref-Lawrence2013">
<p>Lawrence, Michael, Wolfgang Huber, Hervé Pagès, Patrick Aboyoun, Marc Carlson, Robert Gentleman, Martin T. Morgan, and Vincent J. Carey. 2013. “Software for Computing and Annotating Genomic Ranges.” Edited by Andreas Prlic. <em>PLoS Computational Biology</em> 9 (8). Public Library of Science: e1003118+. <a href="https://doi.org/10.1371/journal.pcbi.1003118">https://doi.org/10.1371/journal.pcbi.1003118</a>.</p>
</div>
<div id="ref-Love2019">
<p>Love, Michael I., Charlotte Soneson, Peter F. Hickey, Lisa K. Johnson, N. Tessa Pierce, Lori Shepherd, Martin Morgan, and Rob Patro. 2019. “Tximeta: reference sequence checksums for provenance identification in RNA-seq.” <em>bioRxiv</em>. <a href="https://doi.org/10.1101/777888">https://doi.org/10.1101/777888</a>.</p>
</div>
<div id="ref-Patro2017">
<p>Patro, R, G Duggal, MI Love, RA Irizarry, and C Kingsford. 2017. “Salmon Provides Fast and Bias-Aware Quantification of Transcript Expression.” <em>Nature Methods</em> 14: 417–19. <a href="https://doi.org/10.1038/nmeth.4197">https://doi.org/10.1038/nmeth.4197</a>.</p>
</div>
<div id="ref-Soneson2015">
<p>Soneson, Charlotte, Michael I. Love, and Mark Robinson. 2015. “Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences.” <em>F1000Research</em> 4 (1521). <a href="https://doi.org/10.12688/f1000research.7563">https://doi.org/10.12688/f1000research.7563</a>.</p>
</div>
<div id="ref-Srivastava2019">
<p>Srivastava, Avi, Laraib Malik, Tom Sean Smith, Ian Sudbery, and Rob Patro. 2019. “Alevin efficiently estimates accurate gene abundances from dscRNA-seq data.” <em>Genome Biology</em> 20 (65). <a href="https://doi.org/10.1186/s13059-019-1670-y">https://doi.org/10.1186/s13059-019-1670-y</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="model.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Statistical Modeling of High Dimensional Counts.pdf", "Statistical Modeling of High Dimensional Counts.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
