<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 References | Statistical Modeling of High Dimensional Counts</title>
  <meta name="description" content="This book covers some basic routines for statistical modeling of high dimensional count data in using R/Bioconductor." />
  <meta name="generator" content="bookdown 0.16 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 References | Statistical Modeling of High Dimensional Counts" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book covers some basic routines for statistical modeling of high dimensional count data in using R/Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 References | Statistical Modeling of High Dimensional Counts" />
  
  <meta name="twitter:description" content="This book covers some basic routines for statistical modeling of high dimensional count data in using R/Bioconductor." />
  

<meta name="author" content="Michael I. Love" />


<meta name="date" content="2020-01-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="session.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Stat. Model. of High Dim. Counts</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this online book</a></li>
<li class="chapter" data-level="1" data-path="quantification.html"><a href="quantification.html"><i class="fa fa-check"></i><b>1</b> Quantification</a></li>
<li class="chapter" data-level="2" data-path="model.html"><a href="model.html"><i class="fa fa-check"></i><b>2</b> Counts modeling</a><ul>
<li class="chapter" data-level="2.1" data-path="model.html"><a href="model.html#a-first-exploration-of-counts"><i class="fa fa-check"></i><b>2.1</b> A first exploration of counts</a></li>
<li class="chapter" data-level="2.2" data-path="model.html"><a href="model.html#modeling-counts-with-deseq2"><i class="fa fa-check"></i><b>2.2</b> Modeling counts with DESeq2</a></li>
<li class="chapter" data-level="2.3" data-path="model.html"><a href="model.html#hypothesis-weighting"><i class="fa fa-check"></i><b>2.3</b> Hypothesis weighting</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="transcripts.html"><a href="transcripts.html"><i class="fa fa-check"></i><b>3</b> Transcript expression</a></li>
<li class="chapter" data-level="4" data-path="limits.html"><a href="limits.html"><i class="fa fa-check"></i><b>4</b> Limitations and extensions</a></li>
<li class="chapter" data-level="5" data-path="session.html"><a href="session.html"><i class="fa fa-check"></i><b>5</b> Session information</a></li>
<li class="chapter" data-level="6" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>6</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Modeling of High Dimensional Counts</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> References</h1>

<div id="refs" class="references">
<div>
<p>Alasoo, K., Rodrigues, J., Mukhopadhyay, S., Knights, A., Mann, A., Kundu, K., … Gaffney, D. (2018). Shared genetic effects on chromatin and gene expression indicate a role for enhancer priming in immune response. <em>Nature Genetics</em>, <em>50</em>, 424–431.</p>
</div>
<div>
<p>Al Seesi, S., Temate-Tiagueu, Y., Zelikovsky, A., and Măndoiu, I. I. (2014). Bootstrap-based differential gene expression analysis for RNA-Seq data with and without replicates. <em>BMC Genomics</em>, <em>15</em>(Suppl 8).</p>
</div>
<div>
<p>Anders, S., and Huber, W. (2010). Differential expression analysis for sequence count data. <em>Genome Biology</em>, <em>11</em>, R106.</p>
</div>
<div>
<p>Anders, S., Reyes, A., and Huber, W. (2012). Detecting differential usage of exons from RNA-seq data. <em>Genome Research</em>, <em>22</em>(10), 2008–2017.</p>
</div>
<div>
<p>Benjamini, Y., and Hochberg, Y. (1995). Controlling the false discovery rate: A practical and powerful approach to multiple testing. <em>Journal of the Royal Statistical Society B</em>, <em>57</em>, 289–300.</p>
</div>
<div>
<p>Bray, N. L., Pimentel, H., Melsted, P., and Pachter, L. (2016). Near-optimal probabilistic RNA-seq quantification. <em>Nature Biotechnology</em>, <em>34</em>(5), 525.</p>
</div>
<div>
<p>Calgaro, M., Romualdi, C., Waldron, L., Risso, D., and Vitulo, N. (2020). Assessment of single cell rna-seq statistical methods on microbiome data. <em>bioRxiv</em>.</p>
</div>
<div>
<p>Callahan, B., Sankaran, K., Fukuyama, J., McMurdie, P., and Holmes, S. (2016). Bioconductor Workflow for Microbiome Data Analysis: from raw reads to community analyses. <em>F1000Research</em>, <em>5</em>(1492).</p>
</div>
<div>
<p>Castel, S. E., Levy-Moonshine, A., Mohammadi, P., Banks, E., and Lappalainen, T. (2015). Tools and best practices for data processing in allelic expression analysis. <em>Genome Biology</em>, <em>16</em>(1), 195.</p>
</div>
<div>
<p>Cox, D. R., and Reid, N. (1987). Parameter orthogonality and approximate conditional inference. <em>Journal of the Royal Statistical Society, Series B</em>, <em>49</em>(1), 1–39.</p>
</div>
<div>
<p>Dudoit, S., Yang, Y. H., Callow, M. J., and Speed, T. P. (2002). Statistical methods for identifying differentially expressed genes in replicated cDNA microarray experiments. <em>Statistica Sinica</em>, <em>12</em>(1), 111–139.</p>
</div>
<div>
<p>Duo, A., Robinson, M., and Soneson, C. (2018). A systematic performance evaluation of clustering methods for single-cell RNA-seq data. <em>F1000Research</em>, <em>7</em>(1141).</p>
</div>
<div>
<p>Ewels, P., Magnusson, M., Lundin, S., and Käller, M. (2016). MultiQC: summarize analysis results for multiple tools and samples in a single report. <em>Bioinformatics</em>, <em>32</em>(19), 3047–3048.</p>
</div>
<div>
<p>Fernandes, A. D., Reid, J. N., Macklaim, J. M., McMurrough, T. A., Edgell, D. R., and Gloor, G. B. (2014). Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis. <em>Microbiome</em>, <em>2</em>(1), 15.</p>
</div>
<div>
<p>Frankish, A., GENCODE-consoritum, and Flicek, P. (2018). GENCODE reference annotation for the human and mouse genomes. <em>Nucleic Acids Research</em>.</p>
</div>
<div>
<p>Glaus, P., Honkela, A., and Rattray, M. (2012). Identifying differentially expressed transcripts from RNA-seq data with biological variation. <em>Bioinformatics</em>, <em>28</em>(13).</p>
</div>
<div>
<p>Huber, W., Carey, V. J., Gentleman, R., Anders, S., Carlson, M., Carvalho, B. S., … Morgan, M. (2015). Orchestrating high-throughput genomic analysis with Bioconductor. <em>Nature Methods</em>, <em>12</em>(2), 115–121.</p>
</div>
<div>
<p>Ignatiadis, N., Klaus, B., Zaugg, J., and Huber, W. (2016). Data-driven hypothesis weighting increases detection power in genome-scale multiple testing. <em>Nature Methods</em>.</p>
</div>
<div>
<p>King, H. W., and Klose, R. J. (2017). The pioneer factor oct4 requires the chromatin remodeller brg1 to support gene regulatory element function in mouse embryonic stem cells. <em>eLife</em>, <em>6</em>, e22631.</p>
</div>
<div>
<p>Köster, J., and Rahmann, S. (2012). Snakemake—a scalable bioinformatics workflow engine. <em>Bioinformatics</em>, <em>28</em>(19), 2520–2522.</p>
</div>
<div>
<p>Law, C. W., Chen, Y., Shi, W., and Smyth, G. K. (2014). voom: precision weights unlock linear model analysis tools for RNA-seq read counts. <em>Genome Biology</em>, <em>15</em>(2), 29.</p>
</div>
<div>
<p>Lawrence, M., Huber, W., Pagès, H., Aboyoun, P., Carlson, M., Gentleman, R., … Carey, V. J. (2013). Software for Computing and Annotating Genomic Ranges. <em>PLoS Computational Biology</em>, <em>9</em>(8), e1003118+.</p>
</div>
<div>
<p>Li, H. (2018). Minimap2: pairwise alignment for nucleotide sequences. <em>Bioinformatics</em>, <em>34</em>(18), 3094–3100.</p>
</div>
<div>
<p>Li, J., and Tibshirani, R. (2011). Finding consistent patterns: A nonparametric approach for identifying differential expression in RNA-Seq data. <em>Statistical Methods in Medical Research</em>, <em>22</em>(5), 519–536.</p>
</div>
<div>
<p>Love, M., Anders, S., Kim, V., and Huber, W. (2015). RNA-seq workflow: Gene-level exploratory analysis and differential expression. <em>F1000Research</em>, <em>4</em>(1070).</p>
</div>
<div>
<p>Love, M. I., Hogenesch, J. B., and Irizarry, R. A. (2016). Modeling of rna-seq fragment sequence bias reduces systematic errors in transcript abundance estimation. <em>Nature Biotechnology</em>, <em>34</em>(12), 1287–1291.</p>
</div>
<div>
<p>Love, M. I., Huber, W., and Anders, S. (2014). Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. <em>Genome Biology</em>, <em>15</em>(12), 550.</p>
</div>
<div>
<p>Love, M. I., Soneson, C., Hickey, P. F., Johnson, L. K., Pierce, N. T., Shepherd, L., … Patro, R. (2019). Tximeta: reference sequence checksums for provenance identification in RNA-seq. <em>bioRxiv</em>.</p>
</div>
<div>
<p>Love, M., Soneson, C., and Patro, R. (2018). Swimming downstream: Statistical analysis of differential transcript usage following salmon quantification. <em>F1000Research</em>, <em>7</em>(952).</p>
</div>
<div>
<p>McCarthy, D. J., Chen, Y., and Smyth, G. K. (2012). Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation. <em>Nucleic Acids Research</em>, <em>40</em>, 4288–4297.</p>
</div>
<div>
<p>Nowicka, M., and Robinson, M. (2016). DRIMSeq: a Dirichlet-multinomial framework for multivariate count outcomes in genomics. <em>F1000Research</em>, <em>5</em>(1356).</p>
</div>
<div>
<p>Patro, R., Duggal, G., Love, M., Irizarry, R., and Kingsford, C. (2017). Salmon provides fast and bias-aware quantification of transcript expression. <em>Nature Methods</em>, <em>14</em>, 417–419.</p>
</div>
<div>
<p>Pimentel, H., Bray, N. L., Puente, S., Melsted, P., and Pachter, L. (2017). Differential analysis of RNA-seq incorporating quantification uncertainty. <em>Nature Methods</em>, <em>14</em>(7), 687–690.</p>
</div>
<div>
<p>Raghupathy, N., Choi, K., Vincent, M. J., Beane, G. L., Sheppard, K. S., Munger, S. C., … Churchill, G. A. (2018). Hierarchical analysis of RNA-seq reads improves the accuracy of allele-specific expression. <em>Bioinformatics</em>, <em>34</em>(13), 2177–2184.</p>
</div>
<div>
<p>Roberts, C. J., Nelson, B., Marton, M. J., Stoughton, R., Meyer, M. R., Bennett, H. A., … Friend,. S. H. (2000). Signaling and circuitry of multiple mapk pathways revealed by a matrix of global gene expression profiles. <em>Science</em>, <em>287</em>(5454), 873–880.</p>
</div>
<div>
<p>Robinson, M. D., McCarthy, D. J., and Smyth, G. K. (2010). edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. <em>Bioinformatics</em>, <em>26</em>(1), 139.</p>
</div>
<div>
<p>Sankaran, K., and Holmes, S. P. (2018). Latent variable modeling for the microbiome. <em>Biostatistics</em>, <em>20</em>(4), 599–614.</p>
</div>
<div>
<p>Soneson, C., Love, M. I., and Robinson, M. (2015). Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences. <em>F1000Research</em>, <em>4</em>(1521).</p>
</div>
<div>
<p>Soneson, C., Matthes, K. L., Nowicka, M., Law, C. W., and Robinson, M. D. (2016). Isoform prefiltering improves performance of count-based methods for analysis of differential transcript usage. <em>Genome Biology</em>, <em>17</em>(1), 12.</p>
</div>
<div>
<p>Soneson, C., and Robinson, M. D. (2018). Bias, robustness and scalability in single-cell differential expression analysis. <em>Nature Methods</em>, <em>15</em>(4), 255–261.</p>
</div>
<div>
<p>Soneson, C., Yao, Y., Bratus-Neuenschwander, A., Patrignani, A., Robinson, M. D., and Hussain, S. (2019). A comprehensive examination of Nanopore native RNA sequencing for characterization of complex transcriptomes. <em>Nature Communications</em>, <em>10</em>(1), 3359.</p>
</div>
<div>
<p>Srivastava, A., Malik, L., Sarkar, H., Zakeri, M., Almodaresi, F., Soneson, C., … Patro, R. (2019). Alignment and mapping methodology influence transcript abundance estimation. <em>bioRxiv</em>.</p>
</div>
<div>
<p>Srivastava, A., Malik, L., Smith, T. S., Sudbery, I., and Patro, R. (2019). Alevin efficiently estimates accurate gene abundances from dscRNA-seq data. <em>Genome Biology</em>, <em>20</em>(65).</p>
</div>
<div>
<p>Stephens, M. (2016). False discovery rates: A new deal. <em>Biostatistics</em>, <em>18</em>(2).</p>
</div>
<div>
<p>Storey, J., and Tibshirani, R. (2003).  Statistical significance for genome-wide experiments. <em>Proceedings of the National Academy of Sciences</em>, <em>100</em>(16), 9440–9445.</p>
</div>
<div>
<p>Sun, S., Zhu, J., Ma, Y., and Zhou, X. (2019). Accuracy, Robustness and Scalability of Dimensionality Reduction Methods for Single Cell RNAseq Analysis. <em>bioRxiv</em>.</p>
</div>
<div>
<p>Tibshirani, R. (1988). Estimating transformations for regression via additivity and variance stabilization. <em>Journal of the American Statistical Association</em>, <em>83</em>, 394–405.</p>
</div>
<div>
<p>Townes, F. W., Hicks, S. C., Aryee, M. J., and Irizarry, R. A. (2019). Feature Selection and Dimension Reduction for Single Cell RNA-Seq based on a Multinomial Model. <em>bioRxiv</em>.</p>
</div>
<div>
<p>Turro, E., Astle, W. J., and Tavaré, S. (2013). Flexible analysis of RNA-seq data using mixed effects models. <em>Bioinformatics</em>, <em>30</em>(2), 180–188.</p>
</div>
<div>
<p>Turro, E., Su, S.-Y., Gonçalves, Â., Coin, L. J., Richardson, S., and Lewin, A. (2011). Haplotype and isoform specific expression estimation using multi-mapping RNA-seq reads. <em>Genome Biology</em>, <em>12</em>(2), R13.</p>
</div>
<div>
<p>Van den Berge, K., Hembach, K. M., Soneson, C., Tiberi, S., Clement, L., Love, M. I., … Robinson, M. D. (2019). RNA sequencing data: Hitchhiker’s guide to expression analysis. <em>Annual Review of Biomedical Data Science</em>, <em>2</em>(1), 139–173.</p>
</div>
<div>
<p>Van den Berge, K., Perraudeau, F., Soneson, C., Love, M. I., Risso, D., Vert, J.-P., … Clement, L. (2018). Observation weights unlock bulk RNA-seq tools for zero inflation and single-cell applications. <em>Genome Biology</em>, <em>19</em>(24).</p>
</div>
<div>
<p>Van den Berge, K., Soneson, C., Robinson, M. D., and Clement, L. (2017). stageR: a general stage-wise method for controlling the gene-level false discovery rate in differential expression and differential transcript usage. <em>Genome Biology</em>, <em>18</em>(1), 151.</p>
</div>
<div>
<p>Willis, A. D. (2019). Rarefaction, alpha diversity, and statistics. <em>Frontiers in Microbiology</em>, <em>10</em>, 2407.</p>
</div>
<div>
<p>Witten, D. M. (2011). Classification and clustering of sequencing data using a Poisson model. <em>The Annals of Applied Statistics</em>, <em>5</em>(4), 2493–2518.</p>
</div>
<div>
<p>Wu, H., Wang, C., and Wu, Z. (2012). A new shrinkage estimator for dispersion improves differential expression detection in RNA-seq data. <em>Biostatistics</em>.</p>
</div>
<div>
<p>Zhu, A., Ibrahim, J. G., and Love, M. I. (2018). Heavy-tailed prior distributions for sequence count data: Removing the noise and preserving large differences. <em>Bioinformatics</em>.</p>
</div>
<div>
<p>Zhu, A., Srivastava, A., Ibrahim, J., Patro, R., and Love, M. (2019). Nonparametric expression analysis using inferential replicate counts. <em>Nucleic Acids Research</em>.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="session.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Statistical Modeling of High Dimensional Counts.pdf", "Statistical Modeling of High Dimensional Counts.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
